rancher/rancher
# Backport "Update rancher machine calls (#42995)"

## Issue: https://github.com/rancher/rancher/issues/43574

 
## Problem
Need to backport fix for [this issue](https://github.com/rancher/rancher/issues/40608) to v2.7.
 
## Solution
Backporting [my changes](https://github.com/rancher/rancher/pull/42995) for [the original issue](https://github.com/rancher/rancher/issues/40608) to v2.7.
 
## Testing


## Engineering Testing
### Manual Testing
Created a cluster, rotated cloud credentials, and then made sure that cluster deletion succeeded gracefully, just to make sure I did the backport properly. Other tests for this change were implemented in Rancher Machine when I originally implemented it.

### Automated Testing

See testing section from my original PR: https://github.com/rancher/rancher/pull/42995

## QA Testing Considerations

 
### Regressions Considerations

I think the main thing to focus on is the case where credentials are changed after cluster creation. This is relevant for all node drivers, so we probably want to test this scenario with each one to ensure it works as intended.
